user_sync/rules.py

```diff
@@ -777,38 +870,74 @@ def execute_actions(self):
             if not had_work:
                 break
     
-class Group(object):
-
+class ConfigGroup(object):
     dashboard_groups = {}
 
-    def __init__(self, group_name, organization_name):
+    def __init__(self, group_name, organization_name, designation):
         '''
         :type group_name: str
         :type organization_name: str        
         '''
         self.group_name = group_name
         self.organization_name = organization_name
-        Group.dashboard_groups[(group_name, organization_name)] = self
+        self.designation = designation
+
+        ConfigGroup.dashboard_groups[(group_name, organization_name)] = self
     
+    def key(self):
+        return { 'group_name': self.group_name, 'organization_name': self.organization_name }
+
     def __eq__(self, other):
-        return self.__dict__ == other.__dict__
+        return self.key() == other.key() if other != None else False
 
     def __ne__(self, other):
-        return not self.__eq__(other)
+        return self.key() != other.key() if other != None else True
     
     def __hash__(self):
-        return hash(frozenset(self.__dict__))
+        return hash(frozenset(self.key()))
     
     def __str__(self):
-        return str(self.__dict__)
+        return str(self.key())
 
     @classmethod
     def get_dashboard_group(cls, group_name, organization_name):
         '''
         :type group_name: str
         :type organization_name: str
         '''
-        return Group.dashboard_groups.get((group_name, organization_name))
+        return ConfigGroup.dashboard_groups.get((group_name, organization_name))
+
+class TargetGroup(object):
+    def __init__(self, group_name_or_config_group, organization_name=None, designation=None):
+        '''
+        :type group_name: str
+        :type organization_name: str
+        '''
+        if isinstance(group_name_or_config_group, ConfigGroup):
```

Comment:
I'm normally not fond of conversion like this since TargetGroup has to know about ConfigGroup.  I would have used a helper function to convert ConfigGroup to TargetGroup.